tools/python/gen_opkernel_doc.py

```diff
@@ -44,36 +43,49 @@ def format_param_strings(params):
             if firstparam:
                 firstparam = False
             else:
-                s += ' or '
+                s += '<br><br>or<br><br>'
             s += param
     return s
 
 
-def main(args):  # type: (Type[Args]) -> None
+def expand_providers(provider_filter: [str]):
+    providers = set()
+    if provider_filter:
+        for provider in provider_filter:
+            p = provider.lower()
+            if not p.endswith('executionprovider'):
+                p += 'executionprovider'
+            providers.add(p)
 
-    with io.open(args.output, 'w', newline='', encoding="utf-8") as fout:
-        fout.write('## Supported Operators Data Types\n')
+    return providers
+
+
+def main(output_path: pathlib.Path, provider_filter: [str]):
+
+    providers = expand_providers(provider_filter)
+
+    with io.open(output_path, 'w', newline='', encoding="utf-8") as fout:
+        fout.write('## Supported Operators and Data Types\n')
```

Comment:
In OperatorKernels.md it's not clear if other execution providers are also part of the documentation. Is it possible to have TOC hyperlinks?

Reply:
The TOC stuff in gh-pages is coming from the site generation infrastructure. I could try and replicate that in the python but that requires some logic to also create anchors. Does it need to be part of this checkin? Short term could we just list the EPs covered at the top and add the anchors/links in a separate PR?

Reply:
Not urgent. Can be different PR.